### PR TITLE
Extract service.checkThatUserHasRightsForDirectRelation() method + fix small issues

### PR DESCRIPTION
**Pull request by zenovich**

### CODE REVIEW

File: app/api/groups/groups.go [hunk 1]
```diff
@@ -44,3 +44,35 @@ func checkThatUserOwnsTheGroup(store *database.DataStore, user *database.User, g
...
+func checkThatUserHasRightsForDirectRelation(store *database.DataStore, user *database.User, parentGroupID, childGroupID int64) service.APIError {
+	groupStore := store.Groups()
+
+	var groupData []struct {
+		ID   int64  `gorm:"column:ID"`
+		Type string `gorm:"column:sType"`
+	}
+
+	err := groupStore.OwnedBy(user).
+		WithWriteLock().
```
Comment (smadbe) [inline app/api/groups/groups.go]: don't you have to check that it is run in a transaction?

File: app/database/user_test.go [hunk 2]
```diff
@@ -12,7 +12,9 @@ import (
 // NewMockUser creates a mock user, to be used for testing
 //func NewMockUser(id, selfGroupID, ownedGroupID, defaultLanguageID int64) *User {
```
Comment (smadbe) [inline app/database/user_test.go]: is this line still relevant?

File: app/api/groups/groups.go (hunk 1 above)
Comment (zenovich) [reply app/api/groups/groups.go]: 1. WithWriteLock() panics itself when it is called without a transaction.
2. The method is private (unexported) so I don't think we should have a test for this.

File: app/database/user_test.go (hunk 2 above)
Comment (zenovich) [reply app/database/user_test.go]: fixed

File: app/api/groups/groups.go (hunk 1 above)
Comment (zenovich) [reply app/api/groups/groups.go]: And we have a test that checks WithWriteLock() panics when outside a transaction

**Pull request merged**